src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1568,64 +1568,68 @@ class SetStatementSegment(BaseSegment):
-            "FORCEPLAN",
-            "SHOWPLAN_ALL",
-            "SHOWPLAN_TEXT",
-            "SHOWPLAN_XML",
+        Delimited(
             Sequence(
-                "STATISTICS",
                 OneOf(
-                    "IO",
-                    "PROFILE",
-                    "TIME",
-                    "XML",
+                    Ref("ParameterNameSegment"),
+                    "DATEFIRST",
+                    "DATEFORMAT",
+                    "DEADLOCK_PRIORITY",
+                    "LOCK_TIMEOUT",
+                    "CONCAT_NULL_YIELDS_NULL",
+                    "CURSOR_CLOSE_ON_COMMIT",
+                    "FIPS_FLAGGER",
+                    Sequence("IDENTITY_INSERT", Ref("TableReferenceSegment")),
+                    "LANGUAGE",
+                    "OFFSETS",
+                    "QUOTED_IDENTIFIER",
+                    "ARITHABORT",
+                    "ARITHIGNORE",
+                    "FMTONLY",
+                    "NOCOUNT",
+                    "NOEXEC",
+                    "NUMERIC_ROUNDABORT",
+                    "PARSEONLY",
+                    "QUERY_GOVERNOR_COST_LIMIT",
+                    "RESULT_SET_CACHING",  # Azure Synapse Analytics specific
+                    "ROWCOUNT",
+                    "TEXTSIZE",
+                    "ANSI_DEFAULTS",
+                    "ANSI_NULL_DFLT_OFF",
+                    "ANSI_NULL_DFLT_ON",
+                    "ANSI_NULLS",
+                    "ANSI_PADDING",
+                    "ANSI_WARNINGS",
+                    "FORCEPLAN",
+                    "SHOWPLAN_ALL",
+                    "SHOWPLAN_TEXT",
+                    "SHOWPLAN_XML",
+                    Sequence(
+                        "STATISTICS",
+                        OneOf(
+                            "IO",
+                            "PROFILE",
+                            "TIME",
+                            "XML",
+                        ),
+                    ),
+                    "IMPLICIT_TRANSACTIONS",
+                    "REMOTE_PROC_TRANSACTIONS",
+                    Sequence(
+                        "TRANSACTION",
+                        "ISOLATION",
+                        "LEVEL",
+                    ),
+                    "XACT_ABORT",
+                ),
+                OneOf(
+                    "ON",
+                    "OFF",
+                    Sequence(
+                        Ref("EqualsSegment"),
+                        Ref("ExpressionSegment"),
+                    ),
                 ),
-            ),
-            "IMPLICIT_TRANSACTIONS",
-            "REMOTE_PROC_TRANSACTIONS",
-            Sequence(
-                "TRANSACTION",
-                "ISOLATION",
-                "LEVEL",
-            ),
-            "XACT_ABORT",
-        ),
-        OneOf(
-            "ON",
-            "OFF",
-            Sequence(
-                Ref("EqualsSegment"),
-                Ref("ExpressionSegment"),
             ),
         ),
```

Comment:
Top hint: collapse white space when reviewing this PR

![image](https://user-images.githubusercontent.com/10931297/151884314-643ccc6f-8673-4db4-85b9-54d88f2d4574.png)